beater/api/mux.go

```diff
@@ -81,6 +91,7 @@ func NewMux(
 	fetcher agentcfg.Fetcher,
 	ratelimitStore *ratelimit.Store,
 	sourcemapFetcher sourcemap.Fetcher,
+	otlpHandlers *otlpreceiver.HTTPHandlers,
```

Comment:
I think we should be constructing the handlers in NewMux. Do we need to pass otlpHandlers in?

Reply:
Maybe we do need to pass the `*otel.Consumer` in though. See later comment regarding `NewHTTPHandlers`.

Reply:
Passed the consumer in 51cdb119d